### PR TITLE
Output hash record

### PR DESCRIPTION
**Pull request by apalepu23**
- Associating output hash with task record
- Handling zero value for state delta scenario

### CODE REVIEW

File: contracts/impl/TaskImpl.sol [hunk 1]
```diff
@@ -291,9 +291,9 @@ library TaskImpl {
         // Verify the receipt
         verifyReceipt(state, _scAddr, _taskId, _stateDeltaHash, _gasUsed, msg.sender, _sig);
 
-        // Append the new state delta hash and set the contract's output hash
-        secretContract.stateDeltaHashes.push(_stateDeltaHash);
-        uint hashIndex = secretContract.outputHashes.push(_outputHash) - 1;
+        uint hashIndex = _stateDeltaHash != bytes32(0) ? secretContract.stateDeltaHashes.push(_stateDeltaHash) - 1 :
```
Comment (lenak25) [inline contracts/impl/TaskImpl.sol]: I would name this variable here and in the event definition in a more informative way: `deltaIndex` or `deltaHashIndex`

File: enigma-js/test/Enigma.spec.js [hunk 2]
```diff
@@ -1372,53 +1372,84 @@ describe('Enigma tests', () => {
...
+      const endingWorkerBalance = worker.balance;
+      const endingSenderBalance = parseInt(await enigma.tokenContract.methods.balanceOf(task.sender).call());
+      expect(endingWorkerBalance - startingWorkerBalance).toEqual(gasUsed * task.gasPx);
+      expect(endingSenderBalance - startingSenderBalance).toEqual((task.gasLimit - gasUsed) * task.gasPx);
+      expect(result.events.ReceiptVerified).toBeTruthy();
+    });
+
+    it('should count state deltas', async () => {
+      const count = await enigma.admin.countStateDeltas(scAddr);
+      expect(count).toEqual(1);
```
Comment (lenak25) [inline enigma-js/test/Enigma.spec.js:1428]: I would add a check to verify that the number of the contract's deltas hasn't changed after the call to `compute`

**Review by lenak25 — COMMENTED** [review]
Looks good, added several comments.
One more thing that I think should be added to the Enigma JS lib: enabling the user to verify the task output. Whether by adding a separate function for that or by addressing this as part of the `getTaskResult` function.

**Review by lacabra — APPROVED** [review]
Looks good to me

File: contracts/impl/TaskImpl.sol (hunk 1 above)
Comment (apalepu23) [reply contracts/impl/TaskImpl.sol]: resolved: c0ddcde83e6a04514cd1e05611303c2d1dc1dbcf

File: enigma-js/test/Enigma.spec.js (hunk 2 above)
Comment (apalepu23) [reply enigma-js/test/Enigma.spec.js:1428]: resolved: c0ddcde83e6a04514cd1e05611303c2d1dc1dbcf

Comment (apalepu23) [reply enigma-js/test/Enigma.spec.js:1428]: will resolve the additional task verification comment in a following PR as per our discussion offline!

**Pull request merged**